apps/backend/src/modules/users/user.repository.ts

```diff
@@ -11,52 +16,199 @@ class UserRepository implements Repository {
+				email: user.email,
+				firstName: user.firstName,
+				group: GroupEntity.initializeWithPermissions({
+					id: group.id,
+					key: group.key,
+					name: group.name,
+					permissions: permissions.map((permission) =>
+						PermissionEntity.initialize(permission).toObject(),
+					),
+				}).toObject(),
+				groupId: user.groupId,
+				id: user.id,
+				lastName: user.lastName,
+				passwordHash: user.passwordHash,
+				passwordSalt: user.passwordSalt,
+			});
+		});
 	}
 
 	public async findAll(): Promise<UserEntity[]> {
-		const users = await this.userModel.query().execute();
+		const users = await this.userModel
+			.query()
+			.select([
+				"users.id",
+				"firstName",
+				"lastName",
+				"users.group_id as groupId",
+				"email",
+				"group.id",
+				"group.key as groupKey",
+				"group.name as groupName",
+			])
+			.withGraphJoined("group.permissions")
+			.execute();
 
-		return users.map((user) => UserEntity.initialize(user));
+		return users.map((user) => {
+			const group = user.group as NonNullable<typeof user.group>;
+			const permissions = user.group?.permissions as NonNullable<
+				typeof group.permissions
+			>;
+
+			return UserEntity.initialize({
+				email: user.email,
+				firstName: user.firstName,
+				group: GroupEntity.initializeWithPermissions({
+					id: group.id,
+					key: group.key,
+					name: group.name,
+					permissions: permissions.map((permission) =>
+						PermissionEntity.initialize(permission).toObject(),
+					),
+				}).toObject(),
+				groupId: user.groupId,
+				id: user.id,
+				lastName: user.lastName,
+				passwordHash: user.passwordHash,
+				passwordSalt: user.passwordSalt,
+			});
+		});
 	}
 
 	public async findByEmail(email: string): Promise<null | UserEntity> {
-		const user = await this.userModel.query().where("email", email).first();
+		const user = await this.userModel
+			.query()
+			.where("email", email)
+			.withGraphJoined("group.permissions")
+			.first();
 
-		return user ? UserEntity.initialize(user) : null;
+		if (!user) {
+			return null;
+		}
+
+		const group = user.group as NonNullable<typeof user.group>;
+		const permissions = user.group?.permissions as NonNullable<
+			typeof group.permissions
+		>;
+
+		return UserEntity.initialize({
+			email: user.email,
+			firstName: user.firstName,
+			group: GroupEntity.initializeWithPermissions({
+				id: group.id,
+				key: group.key,
+				name: group.name,
+				permissions: permissions.map((permission) =>
+					PermissionEntity.initialize(permission).toObject(),
+				),
+			}).toObject(),
+			groupId: user.groupId,
+			id: user.id,
+			lastName: user.lastName,
+			passwordHash: user.passwordHash,
+			passwordSalt: user.passwordSalt,
+		});
 	}
 
 	public async findById(id: number): Promise<null | UserEntity> {
-		const user = await this.userModel.query().findById(id);
+		const user = (await this.userModel
+			.query()
+			.findById(id)
+			.withGraphJoined("group.permissions")
+			.first()) as UserModel;
```

Comment:
The user can be null here, so we need to return null if the user is not present, the same for findPasswordDetails. In case of create method, we know for sure that if an error wasn't thrown, we have a user, but for finding by parameters, we don't know that
```suggestion
			.first());
```